marimbabot_vision/scripts/data_augmentation.py

```diff
@@ -0,0 +1,52 @@
+from generate_data import NUM_SAMPLES, NUM_WORKER, OUTPUT_DIR
+from multiprocessing import Pool
+from PIL import Image
+
+import tqdm
+import shutil
+import albumentations as A
+import numpy as np
+import cv2
+
+def apply_transforms(img, transforms = [A.RGBShift()]):
+    """apply given transformations"""
+    transform = A.Compose(transforms)
+       
+    transformed = transform(image=np.asarray(img))
+    transformed_image = transformed["image"]
+
+    return transformed_image
+
+def augment_sample(i):
+    """Generate a augmentations for a given sample and save it to disk"""
+    orig_img_path = f"{OUTPUT_DIR}/{i}/staff_1.png"
+    orig_txt_path = f"{OUTPUT_DIR}/{i}/staff_1.txt"
+    orig_ly_path = f"{OUTPUT_DIR}/{i}/staff_1.ly"
+    img = Image.open(orig_img_path)
+
+    # set of simpler transformations
+    transformations = dict({
+        "blur": [A.Blur(blur_limit=15, always_apply=True)],
+        "median_blur" : [A.MedianBlur(blur_limit=5, always_apply=True)],
+        "glass_blur": [A.GlassBlur(sigma=0.4,always_apply=True)],
+        "zoom_blur": [A.ZoomBlur(always_apply=True)],
+        "image_compression": [A.ImageCompression(quality_lower=60, quality_upper=60 ,always_apply=True)],
+        "downscale": [A.Downscale(scale_min=0.4, scale_max=0.4, always_apply=True, interpolation=cv2.INTER_NEAREST)],
+        "rotate_3": [A.Rotate(limit=[3,3], always_apply=True)],
+        "rotate_6": [A.Rotate(limit=[6,6], always_apply=True)],
+        "translate_x_16": [A.Affine(translate_px={"x":16})],
+        "translate_y_16": [A.Affine(translate_px={"y":16})],
+        "translate_x_y_15": [A.Affine(translate_px={"y":15, "x":15})],
+        })
+
+    for key in transformations.keys():
```

Comment:
Instead of applying transformations individually, I would strongly prefer creating a composition of all augmentations (with `always_apply=False`).